src/main/java/ca/wolfram/beta/symath/VMap.java

```diff
@@ -0,0 +1,24 @@
+package ca.wolfram.beta.symath;
+import java.util.HashMap;
+
+/**
+ * Our own hashmap implementation to warn users when overwriting values
+ */
+public class VMap extends HashMap<String, Double> {
+    @Override
+    public Double put(String key, Double value) {
+        if (containsKey(key))
```

Comment:
If it's something where the value never changes, we can make a method where it checks to make sure the input matches the existing value if it exists, and throw an exception otherwise